shared/reducers/chat.js

```diff
@@ -106,13 +116,14 @@ function reducer (state: Constants.State = initialState, action: Constants.Actio
       return new Constants.StateRecord()
     case 'chat:removeOutboxMessage': {
       const {conversationIDKey, outboxID} = action.payload
+      const messageKey = Constants.messageKey(conversationIDKey, 'outboxIDText', outboxID)
       // $FlowIssue
       return state.update('conversationStates', conversationStates => updateConversation(
         conversationStates,
         conversationIDKey,
         // $FlowIssue
         conversation => conversation.update('messages', messages => messages.filter(m => m.outboxID !== outboxID)
-      )))
+      ))).set('messageMap', state.get('messageMap').filter((v, k) => k !== messageKey))
```

Comment:
we have a messageMap (could move to entities later) that just goes from key to message for simple lookup